dali/python/nvidia/dali/plugin/pytorch.py

```diff
@@ -84,12 +84,18 @@ class DALIGenericIterator(object):
                  Outputs will be returned from iterator as dictionary
                  of those names.
                  Each name should be distinct
-    size : int
+    size : int, default = -1
            Number of samples in the epoch (Usually the size of the dataset).
            Providing -1 means that the iterator will work until StopIteration is raised
            from the inside of iter_setup(). The options `fill_last_batch`, `last_batch_padded` and
            `auto_reset` don't work in such case. It works with only one pipeline inside
            the iterator.
+           Mutually exclusive with `reader_name` argument
+    reader_name : str, default = None
+           Name of the reader which will be queried to the shard size, number of shards and
+           all other properties necessary to count properly the number of relevant and padded
+           samples that iterator needs to deal with. It automatically sets `fill_last_batch` and
+           `last_batch_padded` accordingly to match the reader's configuration
```

Comment:
Idea: can we lookup the readers in the pipeline (we sort of can, look at the information map you query - there should be only entries for readers - but we need a specific call for that), check if there is only one and use it automatically?

If the user doesn't set something like `name="MyBestReader"` for the reader they want to use, the actual reader name is some obfuscated garbage.

Reply:
I imagine the situation when you have two readers - one with data, the second with the noise that is used in one of augmentations (like background talk you want to mix with your real audio training data).
Also I want to preserve a legacy behavior and not surprise user how it works after this change.

Reply:
I just meant that as a special case where you have only one reader and you didn't set the size at all. With more than 1 reader, you would need to be explicit which one to use.

Reply:
I wanted to keep old behavior when none is provided.

Reply:
If you didn't set size it means that you relay on the external source to drive the iterator. I would still stick to the requirement to provide the name of the reader if you want it to drive the iterator.

Reply:
Fine.